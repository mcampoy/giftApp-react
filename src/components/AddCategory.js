import React,  {useState} from 'react';
import PropTypes from 'prop-types';


const AddCategory = ( { setCategories } ) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue, ...categories])
            setInputValue('');
        }
        // console.log('Submit hecho')
    }

    return (
        <form className="form" onSubmit={ handleSubmit }>
            <input
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
                // autoFocus
                placeholder="Escribí aquí tu búsqueda y presioná enter"
                />
                {/* <p>{ inputValue }</p> */}
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
