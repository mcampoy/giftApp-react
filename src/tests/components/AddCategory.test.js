import React from 'react'
import { shallow } from "enzyme";
import AddCategory  from "../../components/AddCategory";


describe('Pruebas en <AddCategory />', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories} /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories} /> );
    })



    test('debe mostrarse correcatamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'hola';

        input.simulate('change', { target: { value } });

        // expect( wrapper.find('p').text().trim() ).toBe( value )
    })

    test('NO debe postear la info on submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {

        const value = 'hola';
        const input = wrapper.find('input');

        input.simulate('change', { target: {value} });

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        
        // expect(input.text()).toBe('')
        expect(input.prop('value')).toBe('')



    })
    


})