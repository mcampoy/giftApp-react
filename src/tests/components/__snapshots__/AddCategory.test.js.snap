// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrarse correcatamente 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Escribí aquí tu búsqueda"
    type="text"
    value=""
  />
  <button
    className="btnSend"
    type="submit"
  >
    <i
      className="fas fa-search"
    />
  </button>
</form>
`;
